source/FreeRTOS_Routing.c

```diff
@@ -1410,27 +1426,50 @@ struct xIPv6_Couple
             ( void ) pxIPAddress;
             return pxNetworkEndPoints;
         }
-    #endif
+
 /*-----------------------------------------------------------*/
 
-    #if ( ipconfigUSE_IPv6 != 0 )
         NetworkEndPoint_t * FreeRTOS_FindEndPointOnNetMask_IPv6( const IPv6_Address_t * pxIPv6Address )
         {
             ( void ) pxIPv6Address;
             return pxNetworkEndPoints;
         }
 
-    #endif
 /*-----------------------------------------------------------*/
 
-    #if ( ipconfigUSE_IPv6 != 0 )
         NetworkEndPoint_t * FreeRTOS_FirstEndPoint_IPv6( const NetworkInterface_t * pxInterface )
         {
             ( void ) pxInterface;
             return pxNetworkEndPoints;
         }
 
-    #endif
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Find an end-point that handles a given IPv4-address.
```

Comment:
If the device endpoint is in the same subnet as the given IP address, return the
endpoint. Otherwise, return NULL.